### PR TITLE
FIO register address on public domain

### PR DESCRIPTION
**Pull request by andreyvEze**

### CODE REVIEW

File: src/modules/FioAddress/components/DomainListModal.js [hunk 1]
```diff
@@ -141,50 +156,50 @@ class DomainListModalConnected extends React.Component<Props, State> {
...
+            label=""
+            onChangeText={this.onSearchFilterChange}
+            onSubmitEditing={() => this.selectCustom(input)}
+            style={formFieldStyles}
+            value={input}
+          />
+        </View>
+        {!items.length && <PrimaryButton label={s.strings.submit} onPress={() => this.selectCustom(input)} marginRem={1} />}
+        <FlatList data={items} initialNumToRender={24} keyboardShouldPersistTaps="handled" keyExtractor={this.keyExtractor} renderItem={this.renderItem} />
+      </ThemedModal>
```
Comment (andreyvEze) [inline src/modules/FioAddress/components/DomainListModal.js:187]: add arrow to close the modal

File: src/components/scenes/FioAddressRegisterScene.js [hunk 2]
```diff
@@ -193,8 +194,27 @@ export class FioAddressRegisterScene extends React.Component<Props, State> {
...
+      if (isCustomDomain) {
+        try {
+          const isDomainPublic = fioPlugin.otherMethods ? await fioPlugin.otherMethods.isDomainPublic(domain) : false
+          if (!isDomainPublic) {
+            throw new FioError(s.strings.fio_address_register_domain_is_not_public, FIO_DOMAIN_IS_NOT_PUBLIC)
+          }
+        } catch (e) {
+          if (!e.labelCode || e.name !== 'FioError') {
+            showError(s.strings.fio_connect_wallets_err)
+            return this.setState({
```
Comment (andreyvEze) [inline src/components/scenes/FioAddressRegisterScene.js:207]: remove

File: src/modules/FioAddress/components/DomainListModal.js [hunk 3]
```diff
@@ -104,17 +106,30 @@ class DomainListModalConnected extends React.Component<Props, State> {
...
   selectItem = (value: any) => this.props.bridge.resolve(value)
   renderItem = ({ item }: FlatListItem<Item>) => {
+    const { theme } = this.props
     const { value, label, createNew } = item
+    const styles = getStyles(theme)
     if (createNew) {
       return (
-        <TouchableHighlight onPress={Actions[Constants.FIO_DOMAIN_REGISTER]} underlayColor={THEME.COLORS.TRANSPARENT}>
+        <TouchableHighlight onPress={this.registerNewDomain} underlayColor="transparent">
           <View style={[styles.rowContainerTop, styles.domainListRowContainerTop]}>
```
Comment (andreyvEze) [inline src/modules/FioAddress/components/DomainListModal.js:128]: check if we can leave only one View

File: src/modules/FioAddress/components/DomainListModal.js [hunk 4]
```diff
@@ -123,7 +138,7 @@ class DomainListModalConnected extends React.Component<Props, State> {
     }
     if (value) {
       return (
-        <TouchableHighlight onPress={() => this.selectItem(value)} underlayColor={THEME.COLORS.TRANSPARENT}>
+        <TouchableHighlight onPress={() => this.selectItem(value)} underlayColor="transparent">
           <View style={[styles.rowContainerTop, styles.domainListRowContainerTop]}>
```
Comment (andreyvEze) [inline src/modules/FioAddress/components/DomainListModal.js:142]: leave one View

File: src/components/scenes/FioAddressRegisterScene.js [hunk 5]
```diff
@@ -204,9 +204,6 @@ export class FioAddressRegisterScene extends React.Component<Props, State> {
         } catch (e) {
           if (!e.labelCode || e.name !== 'FioError') {
             showError(s.strings.fio_connect_wallets_err)
-            return this.setState({
```
Comment (thehobbit85) [inline src/components/scenes/FioAddressRegisterScene.js]: I think you needed to add:
'isAvailable: false'
to setState instead of removing it no?

Comment (andreyvEze) [reply src/components/scenes/FioAddressRegisterScene.js]: no, there is setState below
```
if (!e.labelCode || e.name !== 'FioError') {
            showError(s.strings.fio_connect_wallets_err)
          }
          return this.setState({
            isAvailable: false,
            loading: false
          })
```

**Comment by andreyvEze** [comment]
Waiting for API endpoint on FIO registration site

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**